From 320cddcdf3affa5a0e0604c9422bfdf775dc1f99 Mon Sep 17 00:00:00 2001
From: "Jason A. Donenfeld" <Jason@zx2c4.com>
Date: Tue, 22 Jun 2021 14:10:49 +0200
Subject: [PATCH] common: don't define FMT_USE_USER_DEFINED_LITERALS
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Doing so causes this error:

  In file included from src/common/common.h:39,
                   from src/common/common_pch.h:1:
  /usr/include/fmt/format.h:2752:64: error: ‘udl_arg’ in namespace ‘fmt::v7::detail’ does not name a template type
   2752 | constexpr auto operator"" _a(const char* s, size_t) -> detail::udl_arg<char> {

Signed-off-by: Jason A. Donenfeld <Jason@zx2c4.com>
---
 src/common/common.h | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/common/common.h b/src/common/common.h
index 6bf55b68b..224cbd5a8 100644
--- a/src/common/common.h
+++ b/src/common/common.h
@@ -32,10 +32,6 @@
 
 #include <stdint.h>
 
-// Don't support user-defined literals in fmt as they aren't used by
-// MKVToolNix and produce a compiler warning in -Wpedantic
-#define FMT_USE_USER_DEFINED_LITERALS 0
-
 #include <fmt/format.h>
 #include <fmt/ostream.h>
 
-- 
2.32.0

